### DISCUSSION TITLE
ResourceInUseException: Cannot create preexisting table

### DISCUSSION THREAD

**Discussion by ghost** [dynamoose/dynamoose#1119] [Q&A]
I had created 3 models, Then i deleted those 3 models using docClient.delete, and on recreating all 3 tables using dynamoose models, I get this error "ResourceInUseException: Cannot create preexisting table", but the tables are created too.
 
And then later, while trying to add data to the model i get this error
ValidationException: The table does not have the specified index: departmentNameGlobalIndex

```
var departmentSchema = new dynamoose.Schema({
    departmentId :{type: String,hashKey : true},
    departmentName: {type : String, index:{global:true, rangeKey:'createdAt'}},
    departmentLocation : {type: String, required: true }
},
{
    timestamps : true
});

module.exports = dynamoose.model('Department', departmentSchema)
```

Adding details like this 

```
 Department.query('departmentName').eq(data.departmentName).exec(function (err, result) {
            if (err) {
                console.log(err)
                callback({ "statuscode": 500, "message": "Error while getting department " }) //Getting this error
                return;
            }
            console.log("Department ==>"+JSON.stringify(result))
            if (result != '') {
                callback({ "statuscode": 500, "message": "Department already exists" });
                return;
            }
```

There is some problem i'm facing with secondary index.
Is it like this that the secondary index which was created before were not deleted?

**Comment by fishcharlie** [comment]
`docClient.delete` doesn't delete a table. There also isn't enough information here to fix your issue.

I will note that v2.7.0 of Dynamoose does fix a bug related to indexes being deleted when they shouldn't. So there is a chance that upgrading to v2.7.0 and starting fresh will fix your issue.

Other than that, this sounds a lot more like a bug than a question. So if after trying that it doesn't work, I'd suggest you create a bug report.

**Reply by ghost** [reply]
Okay, I deleted the models and renamed model names (Department to Departments) . Got the same "Cannot create preexisting table",  updated dynamoose to npm update, tables were created and did not face any problem while creating the indexes.

I'm facing another bug, when i updated dynamoose v2.7.0, I'm able to use Model.query but Model.queryOne doesn't work. 
So i copied and pasted the dynamoose from node_modules from a separate project where Model.queryOne was working in v1.1.0.

After pasting in my present project. Model.queryOne was showing after the Model.qu extension in VS code, But while running app.js , I got the error 'q module not found'. So i did npm install and again the queryOne wasn't working.

**Reply by fishcharlie** [reply]
@ssagarGurav Got it.

I didn't realize you were still using v1. Please disregard my message about v2.7.0 fixing that issue. It's obviously unrelated since you are using v1.

v1.1.0 came out in October 2018. **REALLY** long time ago. So I can't help you here. I'd suggest upgrading and staying with the latest versions of software if you want any kind of support.

To start I'd recommend reading the CHANGELOG.md file and looking at the v2.0.0 section. Lots of notes about breaking changes in there. However I realized that I'm not sure it includes the issue you mentioned.

For that issue, `queryOne` is no longer supported. I'd suggest using `query` with a `.limit(1)` instead. Behind the scenes that is exactly what `queryOne` was doing. Just different syntax in v2.

Hopefully this helps.

**Reply by ghost** [reply]
Yes, it worked now. Thank you!
